contracts/src/v0.8/ccip/test/onRamp/OnRamp/MessageTransformerOnRamp.forwardFromRouter.t.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+import {AuthorizedCallers} from "../../../../shared/access/AuthorizedCallers.sol";
+import {IERC20} from "../../../../vendor/openzeppelin-solidity/v4.8.3/contracts/interfaces/IERC20.sol";
+import {NonceManager} from "../../../NonceManager.sol";
+import {Router} from "../../../Router.sol";
+import {IMessageTransformer} from "../../../interfaces/IMessageTransformer.sol";
+import {Client} from "../../../libraries/Client.sol";
+import {Internal} from "../../../libraries/Internal.sol";
+import {MessageTransformerOnRamp} from "../../../onRamp/MessageTransformerOnRamp.sol";
+import {OnRamp} from "../../../onRamp/OnRamp.sol";
+import {MessageTransformerHelper} from "../../helpers/MessageTransformerHelper.sol";
+import {OnRampSetup} from "./OnRampSetup.t.sol";
+
+contract MessageTransformerOnRamp_forwardFromRouter is OnRampSetup {
+  MessageTransformerOnRamp internal s_messageTransformerOnRamp;
+  MessageTransformerHelper internal s_messageTransformer;
+
+  function setUp() public virtual override {
+    super.setUp();
+    s_messageTransformer = new MessageTransformerHelper();
+    s_messageTransformerOnRamp = new MessageTransformerOnRamp(
+      s_onRamp.getStaticConfig(),
+      s_onRamp.getDynamicConfig(),
+      _generateDestChainConfigArgs(s_sourceRouter),
+      address(s_messageTransformer)
+    );
+    s_metadataHash = keccak256(
+      abi.encode(
+        Internal.EVM_2_ANY_MESSAGE_HASH, SOURCE_CHAIN_SELECTOR, DEST_CHAIN_SELECTOR, address(s_messageTransformerOnRamp)
+      )
+    );
+    address[] memory authorizedCallers = new address[](1);
+    authorizedCallers[0] = address(s_messageTransformerOnRamp);
+
+    NonceManager(s_outboundNonceManager).applyAuthorizedCallerUpdates(
+      AuthorizedCallers.AuthorizedCallerArgs({addedCallers: authorizedCallers, removedCallers: new address[](0)})
+    );
+
+    Router.OnRamp[] memory onRampUpdates = new Router.OnRamp[](1);
+    onRampUpdates[0] =
+      Router.OnRamp({destChainSelector: DEST_CHAIN_SELECTOR, onRamp: address(s_messageTransformerOnRamp)});
+
+    Router.OffRamp[] memory offRampUpdates = new Router.OffRamp[](2);
+    offRampUpdates[0] = Router.OffRamp({sourceChainSelector: SOURCE_CHAIN_SELECTOR, offRamp: makeAddr("offRamp0")});
+    offRampUpdates[1] = Router.OffRamp({sourceChainSelector: SOURCE_CHAIN_SELECTOR, offRamp: makeAddr("offRamp1")});
+    s_sourceRouter.applyRampUpdates(onRampUpdates, new Router.OffRamp[](0), offRampUpdates);
+    vm.startPrank(address(s_sourceRouter));
+  }
+
+  function test_forwardFromRouter_WithMessageTransformer_Success() public {
```

Comment:
please follow the new test naming convention for all new tests. See the Foundry guide for the specification. Not all tests in ccip follow it right now but we're slowly replacing them all.

Reply:
yep change function names to follow the convention [here](https://github.com/ScopeLift/scopelint/blob/1857e3940bfe92ac5a136827374f4b27ff083971/src/check/validators/test_names.rs#L106-L143)